contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -3906,9 +3909,29 @@ exec_stmt_change_dbowner(PLtsql_execstate *estate, PLtsql_stmt_change_dbowner *s
 		ereport(ERROR, (errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
 						errmsg("The proposed new database owner is already a user or aliased in the database.")));				
 	}
-					
-	/* All validations done, perform the actual update */
-	update_db_owner(stmt->new_owner_name, stmt->db_name);	
+
+	/* Save the previous user to be restored after granting dbo role to the login. */
+	GetUserIdAndSecContext(&save_userid, &save_sec_context);
+
+	PG_TRY();
+	{
+		/*
+		* Set current user to bbf_role_admin for grant roles.
+		*/
+		SetUserIdAndSecContext(get_bbf_role_admin_oid(), save_sec_context | SECURITY_LOCAL_USERID_CHANGE);
```

Comment:
What if we switch to login(i.e. SessionUser instead)? That will be more secure since bbf_role_admin will have all the privileges but session user will have privileges corresponding only to set of users mapped to it.

Reply:
As discussed offline, we are sticking with the `bbf_role_admin` role to grant roles